From 02c227230a20773cdb616d77d76e0039389df594 Mon Sep 17 00:00:00 2001
From: Lon Hohberger <lhh@redhat.com>
Date: Wed, 30 Jun 2010 11:11:43 -0400
Subject: [PATCH] rgmanager: fix staged upgrade compatibility

This breaks rolling upgrade compatibility for
STABLE3 and should not be ported to that branch.

Signed-off-by: Lon Hohberger <lhh@redhat.com>
---
 rgmanager/include/platform.h |    3 +++
 rgmanager/include/resgroup.h |   22 ++++++++--------------
 2 files changed, 11 insertions(+), 14 deletions(-)

diff --git a/rgmanager/include/platform.h b/rgmanager/include/platform.h
index 7681b50..c5dc66e 100644
--- a/rgmanager/include/platform.h
+++ b/rgmanager/include/platform.h
@@ -47,6 +47,9 @@ Configure is gone...
 #define swab32(x) x=be_swap32(x)
 #define swab64(x) x=be_swap64(x)
 
+#define lswab16(x) x=le_swap16(x)
+#define lswab32(x) x=le_swap32(x)
+#define lswab64(x) x=le_swap64(x)
 
 #if defined(__sparc__)
 #define ALIGNED __attribute__((aligned))
diff --git a/rgmanager/include/resgroup.h b/rgmanager/include/resgroup.h
index 3afbce2..c639315 100644
--- a/rgmanager/include/resgroup.h
+++ b/rgmanager/include/resgroup.h
@@ -28,7 +28,7 @@
 typedef struct {
 	char		rs_name[64];	/**< Service name */
 	/* 64 */
-	uint32_t	rs_id;		/**< Service ID */
+	uint32_t	rs_flags;	/**< Service ID */
 	uint32_t	rs_magic;	/**< Magic ID */
 	uint32_t	rs_owner;	/**< Member ID running service. */
 	uint32_t	rs_last_owner;	/**< Last member to run the service. */
@@ -38,23 +38,17 @@ typedef struct {
 					     restarts */
 	uint64_t	rs_transition;	/**< Last service transition time */
 	/* 96 */
-	uint32_t	rs_flags;	/**< User setted flags */
-	/* 100 */
-	uint8_t		rs_version;	/**< State version */
-	uint8_t		_pad_[3];
-	/* 104 */
 } rg_state_t;
 
 #define swab_rg_state_t(ptr) \
 {\
-	swab32((ptr)->rs_id);\
-	swab32((ptr)->rs_magic);\
-	swab32((ptr)->rs_owner);\
-	swab32((ptr)->rs_last_owner);\
-	swab32((ptr)->rs_state);\
-	swab32((ptr)->rs_restarts);\
-	swab64((ptr)->rs_transition);\
-	swab32((ptr)->rs_flags);\
+	lswab32((ptr)->rs_magic);\
+	lswab32((ptr)->rs_owner);\
+	lswab32((ptr)->rs_last_owner);\
+	lswab32((ptr)->rs_state);\
+	lswab32((ptr)->rs_restarts);\
+	lswab64((ptr)->rs_transition);\
+	lswab32((ptr)->rs_flags);\
 }
 
 #if 0
-- 
1.6.2.5

